client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -18,10 +19,46 @@ import EditorMediaModalDetailPreviewVideo from './detail-preview-video';
 import EditorMediaModalDetailPreviewAudio from './detail-preview-audio';
 import EditorMediaModalDetailPreviewDocument from './detail-preview-document';
 import Button from 'components/button';
-import { userCan, isJetpack } from 'lib/site/utils';
+import { userCan } from 'lib/site/utils';
+import versionCompare from 'lib/version-compare';
 import MediaUtils, { isItemBeingUploaded } from 'lib/media/utils';
 import config from 'config';
 
+/**
+ * This function return true if the image editor can be
+ * enabled/shown
+ *
+ * @param  {object} item - media item
+ * @param  {object} site - current site
+ * @return {boolean} `true` is the image-editor can be enabled.
+ */
+const enableImageEditing = ( item, site ) => {
+	// do not allow if, for some reason, there isn't a valid item yet
+	if ( ! item ) {
+		return false;
+	}
+
+	// do not show if the feature flag isn't set
+	if ( ! config.isEnabled( 'post-editor/image-editor' ) ) {
+		return false;
+	}
+
+	// do not allow for private sites
+	if ( site.is_private ) {
```

Comment:
After bulletproofing this we should be able to 🚢  

I verified behavior on a public wpcom site, private site, Jetpack site, and double checked the version compare behavior.